filter_info.go

```diff
@@ -42,39 +37,48 @@ const (
 
 )
 
-var ErrFieldExist = errors.New("Field is already registered")
+type Context struct {
+	Id               fieldId
+	IdToFieldNameMap map[fieldId]string
+	FieldNameToIdMap map[string]fieldId
+	IdToFieldTypeMap map[fieldId]ftenum
+}
 
-var id fieldId
+var ErrFieldExist = errors.New("gofilter: Field is already registered")
 
 // RegisterField adds field with name and f_type to known fields.
 // When try to register field with name which was already registered
 // return ErrFieldExist.
-func RegisterField(name string, f_type ftenum) error {
-	if exists := fieldNameToIdMap[name]; exists != 0 {
+func (ctx *Context) RegisterField(name string, f_type ftenum) error {

```

Comment:
May be it will be better to keep backward compatible and use the same aproach as in standard package "cmd": FlagSet? 
Something like FilterSet?

https://cs.opensource.google/go/go/+/refs/tags/go1.20.5:src/flag/flag.go;l=1150;drc=9abcc487f8b04db8fd46eaa429a7eef5a5228241

https://cs.opensource.google/go/go/+/refs/tags/go1.20.5:src/flag/flag.go;l=1164;drc=9abcc487f8b04db8fd46eaa429a7eef5a5228241

Reply:
Implemented, needs some documentation though.